packages/time-utilities/src/lib/Cron.ts

```diff
@@ -0,0 +1,138 @@
+/* eslint-disable @typescript-eslint/restrict-plus-operands */
+import { Time, partRegex, wildcardRegex, allowedNum, predefined, cronTokens, tokensRegex } from './constants';
+
+/**
+ * Handles Cron strings and generates dates based on the cron string provided.
+ * @see https://en.wikipedia.org/wiki/Cron
+ */
+export class Cron {
+	public cron: string;
+	public normalized: string;
+	public minutes: number[];
+	public hours: number[];
+	public days: number[];
+	public months: number[];
+	public dows: number[];
+
+	/**
+	 * @param cron The cron pattern to use
+	 */
+	public constructor(cron: string) {
+		this.cron = cron.toLowerCase();
+		this.normalized = Cron.normalize(this.cron);
+		[this.minutes, this.hours, this.days, this.months, this.dows] = Cron.parseString(this.normalized);
+	}
+
+	/**
+	 * Get the next date that matches with the current pattern
+	 * @param outset The Date instance to compare with
+	 * @param origin Whether this next call is origin
+	 */
+	public next(outset: Date = new Date(), origin = true): Date {
+		if (!this.days.includes(outset.getUTCDate()) || !this.months.includes(outset.getUTCMonth() + 1) || !this.dows.includes(outset.getUTCDay())) {
+			return this.next(new Date(outset.getTime() + Time.Day), false);
+		}
+		if (!origin) return new Date(Date.UTC(outset.getUTCFullYear(), outset.getUTCMonth(), outset.getUTCDate(), this.hours[0], this.minutes[0]));
+
+		const now = new Date(outset.getTime() + 60000);
+
+		for (const hour of this.hours) {
+			if (hour < now.getUTCHours()) continue;
+			for (const minute of this.minutes) {
+				if (hour === now.getUTCHours() && minute < now.getUTCMinutes()) continue;
+				return new Date(Date.UTC(outset.getUTCFullYear(), outset.getUTCMonth(), outset.getUTCDate(), hour, minute));
+			}
+		}
+
+		return this.next(new Date(outset.getTime() + Time.Day), false);
+	}
+
+	/**
+	 * Normalize the pattern
+	 * @param cron The pattern to normalize
+	 */
+	private static normalize(cron: string): string {
+		if (cron in predefined) return predefined[cron];
```

Comment:
Use Reflect.has

Reply:
TS will complain coz it doesn't understand Reflect.has for resolving types.